src/Estimators/include/BipedalLocomotion/RobotDynamicsEstimator/UkfState.h

```diff
@@ -0,0 +1,227 @@
+/**
+ * @file UkfState.h
+ * @authors Ines Sorrentino
+ * @copyright 2023 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_ESTIMATORS_UKF_STATE_H
+#define BIPEDAL_LOCOMOTION_ESTIMATORS_UKF_STATE_H
+
+#include <Eigen/Dense>
+#include <memory>
+#include <string>
+
+// BayesFilters
+#include <BayesFilters/AdditiveStateModel.h>
+
+// BLF
+#include <BipedalLocomotion/RobotDynamicsEstimator/Dynamics.h>
+#include <BipedalLocomotion/RobotDynamicsEstimator/SubModel.h>
+#include <BipedalLocomotion/RobotDynamicsEstimator/SubModelKinDynWrapper.h>
+#include <BipedalLocomotion/System/VariablesHandler.h>
+
+namespace BipedalLocomotion
+{
+namespace Estimators
+{
+namespace RobotDynamicsEstimator
+{
+
+/**
+ * UkfState is a concrete class that represents the State of the estimator.
+ * The user should build the dynamic model of the state setting a variable handler
+ * describing the variables composing the state, the list of the dynamic model associated
+ * to each variable in the variable handler, and the matrix of covariances
+ * associated to the state. The user should set also a ukf input provider
+ * which provides the inputs needed to update the ukf process dynamics.
+ */
+class UkfState : public bfl::AdditiveStateModel
+{
+    /**
+     * Private implementation
+     */
+    struct Impl;
+    std::unique_ptr<Impl> m_pimpl;
+
+    //    void unpackState();
+    //    void updateJointAccelerationState();
+
+public:
+    /**
+     * Constructor.
+     */
+    UkfState();
+
+    /**
+     * Destructor.
+     */
+    virtual ~UkfState();
+
+    /**
+     * Build the ukf state model
+     * @param kinDyn a pointer to an iDynTree::KinDynComputations object that will be shared among
+     * all the dynamics.
+     * @param subModelList a vector of pairs (SubModel, SubModelKinDynWrapper) that will be shared
+     * among all the dynamics.
+     * @param handler pointer to the IParametersHandler interface.
+     * @note the following parameters are required by the class
+     * |         Parameter Name         |       Type      | Description | Mandatory |
```

Comment:
Could you make the table more readable? 

He if you want to switch off clang-format for one part of the documentation 
https://releases.llvm.org/3.7.1/tools/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code